specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,318 @@
+                            text += args.InitiatingOrigin;
+                        }
+                        text += " to ";
+                        text += args.Uri;
+                        text += "\n";
+                        text += "Do you want to grant permission?";
+                        string caption = "Launching External Protocol request";
+                        MessageBoxButton btnMessageBox = MessageBoxButton.YesNoCancel;
+                        MessageBoxImage icnMessageBox = MessageBoxImage.None;
+                        MessageBoxResult resultbox = MessageBox.Show(text, caption, btnMessageBox, icnMessageBox);
+                        switch (resultbox)
+                        {
+                            case MessageBoxResult.Yes:
+                                ProcessStartInfo info = new ProcessStartInfo
+                                {
+                                    FileName = args.Uri,
+                                    UseShellExecute = true
+                                };
+                                Process.Start(info);
+                                break;
+
+                            case MessageBoxResult.No:
+                                break;
+
+                            case MessageBoxResult.Cancel:
+                                break;
+                        }
+
+                    }
+                    
+                }
+            }, null);
+        }
+    };
+}
+
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+    
+```IDL
+// This is the ICoreWebView2_16 interface.
+[uuid(cc39bea3-d6d8-471b-919f-da253e2fbf03), object, pointer_default(unique)] 
+interface ICoreWebView2_16 : ICoreWebView2_15 {
+  /// Add an event handler for the `LaunchingExternalProtocol` event.
+  /// The `LaunchingExternalProtocol` event is raised when a launch request is made to
+  /// a protocol that is registered with the OS. 
+  /// The `LaunchingExternalProtocol` event may suppress the default dialog
+  /// or replace the default dialog with a custom dialog.
+  ///
+  /// If a deferral is not taken on the event args, the external protocol launch is 
+  /// blocked until the event handler returns.  If a deferral is taken, the
+  /// external protocol launch is blocked until the deferral is completed.
+  /// The host also has the option to cancel the protocol launch.
+  ///
+  /// The `NavigationStarting` and `NavigationCompleted` events will be raised,
+  /// regardless of whether the `Cancel` property is set to `TRUE` or
+  /// `FALSE`. The `SourceChanged`, `ContentLoading`, and `HistoryChanged` events 
+  /// will not be raised regardless of this property. 
+  /// The `LaunchingExternalProtocol` event will be raised after the 
+  /// `NavigationStarting` event and before the `NavigationCompleted` event.
+  /// The default settings will also be updated upon navigation to an external

```

Comment:
```suggestion
  /// The default settings will also be updated upon navigation to an external

```

I don't know what this means. What default settings? Does this mean that I have to re-register for the event after every navigation? I don't think so.

Reply:
This is the default `CoreWebView2Settings`. This doesn't mean the dev would have to re-register for the event after every navigation.  This only means that if a `CoreWebView2Setting` was updated and a navigation event hasn't been triggered since the update, navigating to an external protocol will update the `CoreWebView2Settings`. I will update this to be clearer.

Reply:
I did not expect my CoreWebView2Settings to change as the result of an attempt to navigate to an external protocol.

I assumed I could do coreWebView2.Settings.AreBrowserAcceleratorKeysEnabled = false, and the setting will remain false until I change it myself. I didn't expect it to return to the default of `true` under certain conditions that didn't exist when I wrote my code originally.

Also, some CoreWebView2Settings properties do not take effect until the next navigation. Is an attempted protocol launch considered a navigation for the purpose of properties like IsGeneralAutofillEnabled? Or do they only take effect on the next *successful* navigation?

Reply:
Please reword to make clear this counts as a navigation for the CoreWebView2Settings that say they apply after the next navigate.